File: Initial Prototype/Client/MyApplicationProject/MyApplication/src/main/java/com/example/myapplication/domain_objects/JourneyMessage.java
package com.example.myapplication.domain_objects;

import java.lang.reflect.Array;
import java.util.ArrayList;

/**
 * Created by Michal on 07/01/14.
 */
public class JourneyMessage {

    public int JourneyMessageId;
    public int JourneyId;
    public int SenderId;
    public String SenderUsername;
    public String MessageBody;
    public String SentOnDate;
    public ArrayList<User> SeenBy;

    public JourneyMessage(int journeyId, String senderUsername, int senderId, String messageBody, String sentOnDate) {
        this.JourneyId = journeyId;
        this.SenderId = senderId;
        this.SenderUsername = senderUsername;
        this.MessageBody = messageBody;
        this.SentOnDate = sentOnDate;
    }
}

File: Project/WebService/DomainObjects/Domains/JourneyMessage.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="Message.cs" company="">
//   
// </copyright>
// <summary>
//   The car share message.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace DomainObjects.Domains
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Runtime.Serialization;

    /// <summary>
    /// The car share message.
    /// </summary>
    [DataContract]
    public class JourneyMessage
    {
        /// <summary>
        /// Gets or sets the car share message id.
        /// </summary>
        [DataMember]
        [ConcurrencyCheck]
        public int JourneyMessageId { get; set; }

        /// <summary>
        /// Gets or sets the car share id.
        /// </summary>
        [DataMember]
        public int JourneyId { get; set; }

        /// <summary>
        /// Gets or sets the sender id.
        /// </summary>
        [DataMember]
        public int SenderId { get; set; }

        /// <summary>
        /// Gets or sets the sender username.
        /// </summary>
        [DataMember]
        public string SenderUsername { get; set; }

        /// <summary>
        /// Gets or sets the message body.
        /// </summary>
        [DataMember]
        public string MessageBody { get; set; }

        /// <summary>
        /// Gets or sets the sent on date.
        /// </summary>
        [DataMember]
        [DataType(DataType.Date)]
        [Column(TypeName = "DateTime2")]
        public DateTime SentOnDate { get; set; }

        /// <summary>
        /// Gets or sets the seen by.
        /// </summary>
        public virtual ICollection<User> SeenBy { get; set; }  
    }
}
